### ISSUE TITLE
Filtering: AND/OR clause

### ISSUE THREAD

**Issue by negati-ve** [strapi/strapi#6139]
**Describe the bug**

Strapi documentation describes filtering with an OR clause as follows

> # Or clauses
> 
> If you use the same operator (except for in and nin) the values will be used to build an OR query
> 
> GET /restaurants?name_contains=pizza&name_contains=giovanni

This does not seem to be true when filter has same operator but two different fields,
resulting in an AND clause (or alteast the output suggests so)

**Expected behavior**
If use of the same operator for different fields and it results in an OR clause,
GET `/uber_rides?requestor_name_eq=adam&responder_name_eq=giovanni`
should output  a set of 
- all uber rides requested by adam (irrespective of who was the responder)
- all uber rides responded by giovanni (irrespective of who was the requestor)

**Observed behavior**
but instead the result for above example is  
- all uber rides requested by adam AND responded by giovanni
in other words, the output results will contain uber rides where (adman was a requestor and giovanni was the responder)

**Additional context**
I still haven't looked under the hood for this issue as this is low priority and can be easily worked around with a custom query
Once i have some time, I will look at this issue a lil more and update accordingly 

Observed on strapi 3.0.0-beta.20 
Database: mysql

**Comment by fastengineer** [comment]
I lost 2 days by dealing with filters and could not find a solution. It's really sad story.

**Comment by derrickmehaffy** [comment]
duplicate: https://github.com/strapi/strapi/issues/3194
